deps/hiredis/dict.c

```diff
@@ -318,14 +318,17 @@ static int _dictExpandIfNeeded(dict *ht) {
 
 /* Our hash table capability is a power of two */
 static unsigned long _dictNextPower(unsigned long size) {
-    unsigned long i = DICT_HT_INITIAL_SIZE;
-
+    if (size <= DICT_HT_INITIAL_SIZE) return DICT_HT_INITIAL_SIZE;
     if (size >= LONG_MAX) return LONG_MAX;
-    while(1) {
-        if (i >= size)
-            return i;
-        i *= 2;
-    }
+    size--;
+    size |= size >> 1;
+    size |= size >> 2;
+    size |= size >> 4;
+    size |= size >> 8;
+    size |= size >> 16;
+    size |= size >> 32;
+    size++;
```

Comment:
I was confused first, but now I realize it's a famous trick. It is described here:

* https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
* https://jameshfisher.com/2018/03/30/round-up-power-2/